lib/shanty/config.rb

```diff
@@ -0,0 +1,51 @@
+require 'shenanigans/hash/to_ostruct'
+require 'deep_merge'
+
+module Shanty
+  # Public: Configuration class for shanty
+  class Config
+    CONFIG_FILE = '.shanty.yml'
+
+    def initialize(root, environment)
+      @root = root
+      @environment = environment
+    end
+
+    def merge!(new_config)
+      @config = config_hash.clone.deep_merge(new_config)
```

Comment:
Rathan than clone, it's probably just as (or more) efficient and a bit terser to do this:

```ruby
{}.deep_merge(config_hash, new_config)
```